bukkit/latest/src/main/java/io/github/wysohn/triggerreactor/bukkit/manager/trigger/share/api/placeholder/PlaceholderExpansionSupport.java

```diff
@@ -93,15 +94,31 @@ public String getVersion() {
      */
     @Override
     public String onPlaceholderRequest(Player player, String identifier) {
-
-        //%tr_version% -> this should return TR version, but should use PluginDescription which is modified as protected method.
-
-
         if (identifier.toLowerCase().equals("version")) {
             return plugin.getVersion();
         }
+        //%tr_?<variable name>% - temporary global variable
+        if (identifier.substring(0, 1).equals("?")) {
```

Comment:
`identifier` can be empty String (when simply %tr_% is used), so it will cause Exception in that case.

Check the length of `identifier` before using substring. I'm pretty sure `identifier` is never null.